### PR TITLE
GH-7414: Moved `fuzzy-search` to common.

### LINKED ISSUE #7414: Move fuzzy search from browser to common.
### Reproduction Steps

**OS and Theia version:**

**Diagnostics:**

### PR DESCRIPTION
**Pull request by kittaakos**
Closes #7414.

Signed-off-by: Akos Kitta <kittaakos@typefox.io>

#### What it does

 - Moves the `fuzzy-search` module from `@theia/core/lib/browser/tree` to `@theia/core/lib/common`.
 - Makes `filter` sync.
 - Fixes a flaw in the range validation. Args were not used.

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Comment by akosyakov** [comment]
I think it is fine I assume it won't break anyone since it was quite internal to trees.

**Comment by akosyakov** [comment]
@kittaakos What about bindings? Do you want to bind it the backend to reuse it there?

File: packages/core/src/browser/tree/fuzzy-search.ts [hunk 1]
```diff
@@ -14,123 +14,25 @@
...
-
-}
+import { FuzzySearch as CommonFuzzySearch } from '../../common/fuzzy-search';
 
 /**
- * Fuzzy searcher.
+ * @deprecated import from `packages/core/src/common/fuzzy-search` instead.
  */
+@injectable()
+export class FuzzySearch extends CommonFuzzySearch { }
```
Comment (akosyakov) [inline packages/core/src/browser/tree/fuzzy-search.ts]: added a comment to the commit by the mistake: https://github.com/eclipse-theia/theia/commit/fbdd35a0227706424c15fe9b3b3c51e63110016b#r38021579

**Comment by kittaakos** [comment]
> Do you want to bind it the backend to reuse it there?

No. I want to do `new FuzzySearch().filter` but in the final electron app we wipe the `browser` folder.

File: packages/core/src/browser/tree/fuzzy-search.ts (hunk 1 above)
Comment (kittaakos) [reply packages/core/src/browser/tree/fuzzy-search.ts]: OK, nevermind.

**Pull request closed without merging**